wgpu-native/src/command/mod.rs

```diff
@@ -313,22 +405,40 @@ pub fn command_encoder_begin_render_pass(
     let render_pass = match render_pass_cache.entry(rp_key.clone()) {
         Entry::Occupied(e) => e.into_mut(),
         Entry::Vacant(e) => {
+            let attachment_unused: u32 = !0; // TODO: Get from ash or expose in gfx-hal. Very important this remains a u32.
+
             let color_ids = [
                 (0, hal::image::Layout::ColorAttachmentOptimal),
                 (1, hal::image::Layout::ColorAttachmentOptimal),
                 (2, hal::image::Layout::ColorAttachmentOptimal),
                 (3, hal::image::Layout::ColorAttachmentOptimal),
             ];
+
+            let mut resolve_ids = vec!();
+            let mut attachment_index = color_attachments.len();
+            if resolve_sample_counts.len() != 0 {
+                for (i, at) in color_attachments.iter().enumerate() {
+                    if at.resolve_target == ptr::null() {
+                        resolve_ids.push((attachment_unused as usize, hal::image::Layout::ColorAttachmentOptimal));
+                    }
+                    else {
+                        assert!(attachment_sample_counts[i] > 1, "RenderPassColorAttachmentDescriptor with a resolve_target must have an attachment with sample_count > 1");
```

Comment:
similarly, I don't think we need to collect that vector separately: we can always get the sample count from the current element of `color_attachments`